.github/workflows/release.yaml

```
@@ -6,18 +6,34 @@
 name: Publish Python Package to PyPI
 on:
   push:
-    tags:
-      - '*.*.*'
+    branches:
+      - main
 
 jobs:
   build_and_publish:
     name: Publish a new version of pypdf
     runs-on: ubuntu-latest
+    if: "startsWith(${{ github.event.head_commit.message}}, 'REL: ')"
```

```suggestion
    if: "${{ startsWith(github.event.head_commit.message, 'REL: ') }}"
```

The old condition will always run this workflow, which clearly is wrong and fails when trying to extract the version number.